### PR TITLE
Feat/grpc api refinements

### PR DESCRIPTION
**Pull request by kepkap**
1 api error handling and other improvements
2 detect tollgate crossing persistence but mostly refactoring

### CODE REVIEW

File: internal/service/location/controller.go [hunk 1]
```diff
@@ -109,30 +123,41 @@ func (c *Controller) UpdateLocation(ctx context.Context, request *locationV1beta
 			To:        to,
 		}
 
-		crossing, err := c.detector.DetectCrossing(ctx, movement)
+		detected, err := c.detector.DetectCrossing(ctx, movement)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("DetectCrossing error: %s", err)
```
Comment (kevinmichaelchen) [inline internal/service/location/controller.go:128]:
```suggestion
			return nil, fmt.Errorf("DetectCrossing error: %v", err)
```

File: internal/service/location/controller.go [hunk 2]
```diff
@@ -109,30 +125,41 @@ func (c *Controller) UpdateLocation(ctx context.Context, request *locationV1beta
...
+		Id:         c.ID,
+		TollgateId: c.TollgateID,
+		WorkerId:   c.WorkerID,
+		Direction:  string(c.Crossing.Crossing.Direction),
+		Alg:        string(c.Crossing.Crossing.Alg),
+		Movement: &locationV1beta1.Movement{
+			FromLon: c.Crossing.Crossing.Movement.From.Lon,
+			FromLat: c.Crossing.Crossing.Movement.From.Lat,
+			ToLon:   c.Crossing.Crossing.Movement.To.Lon,
+			ToLat:   c.Crossing.Crossing.Movement.To.Lat,
```
Comment (kevinmichaelchen) [inline internal/service/location/controller.go:156]: I'd prefer getters where possible. You get a lot of safety in terms of null panics.

File: internal/service/location/controller.go [hunk 3]
```diff
@@ -194,5 +221,5 @@ func (c *Controller) QueryLocation(ctx context.Context, request *locationV1beta1
 			},
 		}, nil
 	}
-	return nil, fmt.Errorf("location not found for WorkerId=%s", request.WorkerId)
+	return nil, status.Errorf(codes.NotFound, "WorkerId: %s", request.WorkerId)
```
Comment (kevinmichaelchen) [inline internal/service/location/controller.go:224]: nit: My preference here is that the error message should be something static (like "Worker not found") and high-cardinality fields (like IDs) should be kept in error metadata. (See #64)

File: internal/service/location/controller_test.go [hunk 4]
```diff
@@ -125,7 +126,8 @@ func testQueryLocation(t *testing.T, client locationV1beta1.LocationServiceClien
 
 	_, err := client.QueryLocation(ctx, request)
 	require.Error(t, err)
-	require.Contains(t, err.Error(), fmt.Sprintf("location not found for WorkerId=%s", request.WorkerId))
+	require.Contains(t, err.Error(), "NotFound")
+	require.Contains(t, err.Error(), request.WorkerId)
```
Comment (kevinmichaelchen) [inline internal/service/location/controller_test.go:130]: You can assert that the status code is `codes.NotFound`

File: internal/service/location/storage/storage.go [hunk 5]
```diff
@@ -58,8 +60,8 @@ func (s *Storage) LastLocation(ctx context.Context, areaKey string, workerID str
 	if len(v) > 0 && v[0] != nil {
 		return &LastLocation{
 			WorkerID:     workerID,
-			Longitude:    v[0].Longitude,
-			Latitude:     v[0].Latitude,
+			Longitude:    util.Round6(v[0].Longitude),
+			Latitude:     util.Round6(v[0].Latitude),
```
Comment (kevinmichaelchen) [inline internal/service/location/storage/storage.go:64]: Why not keep all decimals for maximum precision?

File: internal/service/location/controller.go (hunk 2 above)
Comment (kepkap) [reply internal/service/location/controller.go:156]: thought about that too,  when tollgate_crossing created/persisted Crossing struct is filled completely. so we are safe here, but overall such chaining smells and looks ugly. agreed.

File: internal/service/location/storage/storage.go (hunk 5 above)
Comment (kepkap) [reply internal/service/location/storage/storage.go:64]: round6 is just for clarity, it breaks nothing, but easier for human eye. i spotted 6 digits round used by LatLon converters (somewhere in the web), also google maps seem to limit it to even 5 digits.

**Review by vlkolmakov — APPROVED** [review]

**Pull request merged**